trin-core/src/utp/trin_helpers.rs

```diff
@@ -0,0 +1,107 @@
+// The code here isn't affiliated with stream and isn't part of the uTP implementation
+// this code is used in handling the uTP responses handled in events.rs.
+// just a clarification. It is just for handling the uTP stream higher up what stream releases.
+
+use ssz_derive::{Decode, Encode};
+
+// These Utp impl are related to sending messages over uTP not the implementation itself or stream
+pub struct UtpMessage {
+    pub length: u32,
+    pub payload: Vec<u8>,
+}
+
+impl UtpMessage {
+    pub fn new(payload: Vec<u8>) -> Self {
+        UtpMessage {
+            length: payload.len() as u32,
+            payload,
+        }
+    }
+
+    pub fn encode(&mut self) -> Vec<u8> {
+        let mut buf = vec![];
+        buf.extend_from_slice(&self.length.to_be_bytes());
+        buf.append(&mut self.payload);
+        buf
+    }
+
+    pub fn decode(bytes: &[u8]) -> Result<UtpMessage, String> {
+        if bytes.len() >= 4 {
+            let len = u32::from_be_bytes([bytes[0], bytes[1], bytes[2], bytes[3]]);
```

Comment:
nitpick: Seems like there ought to be some way to pass in some equivalent of `bytes[0..4]` instead of manually entering every item into an array.

Reply:
I wish 😁. That was actually the first thing I tried. We can either do this or I forget what other command, but I believe this ends up being shorter, I would have to check again.

Reply:
How about `try_into()`? Something like:

```suggestion
            let len = u32::from_be_bytes(bytes[0..4].try_into().unwrap()]);
```

Reply:
Ok I changed it to that ty